app/controllers/sponsors_controller.rb

```diff
@@ -37,6 +36,14 @@ def create
   end
 
   def update
+    @sponsor = SponsorAcceptedSlackNotifier.new(
```

Comment:
you can take this even a few steps further. You know what I'm going to say, this controller method knows too much. It should pass current_user, params :id and :sponsor to a factory/create method that returns the object you want in the state you want :)